training/model_management/components/import_model/spec.yaml

```diff
@@ -0,0 +1,117 @@
+
+# Pipeline inputs
+inputs:
+  # pipeline specific compute
+  compute:
+    type: string
+    description: Compute to run pipeline job
+
+  ## Inputs for download model
+  model_source:
+    type: string
+    description: Storage containers from where model will be sourced from.
+    default: Huggingface
+    enum:
+      - AzureBlob
+      - GIT
+      - Huggingface
+
+  model_id:
+    type: string
+    description: A valid model id for the model source selected. For example you can specify `bert-base-uncased` for importing HuggingFace bert base uncased model. Please specify the complete URL if **GIT** or **AzureBlob** is selected in `model_source`
+
+  ## Inputs for the MlFLow conversion
+  license_file_path:
+    type: uri_file
+    description: Path to the license file
+    optional: true
+
+  ## Inputs for Model registration
+  custom_model_name:
+    type: string
+    optional: true
+    description: Model name to use in the registration. If name already exists, the version will be auto incremented
+
+  model_version:
+    type: string
+    optional: true
+    description: Model version in workspace/registry. If the same model name and version exists, the version will be auto incremented
+
+  model_description:
+    type: string
+    optional: true
+    description: Description of the model that will be shown in AzureML registry or workspace
+
+  registry_name:
+    type: string
+    optional: true
+    description: Name of the AzureML asset registry where the model will be registered. Model will be registered in a workspace if this is unspecified
+
+  model_metadata:
+    type: uri_file
+    optional: true
+    description: A JSON or a YAML file that contains model metadata confirming to Model V2 [contract](https://azuremlschemas.azureedge.net/latest/model.schema.json)
+
+# Pipeline outputs
+outputs:
+  mlflow_model_folder:
+    description: Output path for the converted MLFlow model.
+    type: mlflow_model
+
+  model_registration_details:
+    description: Output file which captures transformations applied above and registration details in JSON file.
+    type: uri_file
+
+jobs:
+  download_model:
+    component: azureml:download_model:0.0.1
+    compute: ${{parent.inputs.compute}}
+    inputs:
+      model_source: ${{parent.inputs.model_source}}
+      model_id: ${{parent.inputs.model_id}}
+    outputs:
+      model_download_metadata:
+        type: uri_file
+      model_output:
+        type: uri_folder
+
+  convert_model_to_mlflow:
+    component: azureml:convert_model_to_mlflow:0.0.1
+    compute: ${{parent.inputs.compute}}
+    inputs:
+      license_file_path: ${{parent.inputs.license_file_path}}
+      model_download_metadata: ${{parent.jobs.download_model.outputs.model_download_metadata}}
+      model_path: ${{parent.jobs.download_model.outputs.model_output}}
+    outputs:
+      mlflow_model_folder: ${{parent.outputs.mlflow_model_folder}}
+      model_import_job_path:
+        type: uri_file
+
+  register_model:
```

Comment:
Shouldn't we add identity type in register_model ?

Reply:
Added at pipeline level